src/main/java/de/tum/cit/aet/artemis/communication/dto/ReactionDTO.java

```diff
@@ -2,11 +2,15 @@
 
 import java.time.ZonedDateTime;
 
+import com.fasterxml.jackson.annotation.JsonInclude;
+
 import de.tum.cit.aet.artemis.communication.domain.Reaction;
 
-public record ReactionDTO(Long id, AuthorDTO user, ZonedDateTime creationDate, String emojiId) {
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public record ReactionDTO(Long id, AuthorDTO user, ZonedDateTime creationDate, String emojiId, Long postId, Long answerPostId) {
```

Comment:
Can we please use primitive types for the id's? They should never be null anyways, should they? (see the server guidelines, we prefer primitive values wherever possible https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html#use-dtos-for-efficient-data-transfer) 

Can we also make this information explicit for the other information? 

It seems like user and creationDate are @Nullable?

How about the Emoji? Can it be @Nullable?

```suggestion
public record ReactionDTO(long id, AuthorDTO user, ZonedDateTime creationDate, String emojiId, long postId, Long answerPostId) {
```

Reply:
The user is set in createReaction from userRepository.getUserWithGroupsAndAuthorities() so its also nullable